### ISSUE TITLE
RFE:  Expand Opfor/Ally Random Unit generation and allow to attach modifiers to units.

### ISSUE THREAD

**Issue by Thom293** [MegaMek/mekhq#3578]
TLDR:  Allow StratCon generation parameters to include a means where you can set some more exact stats for a particular unit/force, and allow individual and shared modifiers to be attached to units that will either apply to just that unit or its whole team. (Kind of like facilities have local and shared modifiers) that are separate from the overall global modifiers.  This will allow creation of say a single Commander or Ace or Social General or Tech Stealing a Mech or command lance, etc that has better or worse skills or equipment than the rest of their team and/or may apply bonuses to the rest of the team.  This will be useful for Stratcon now and perhaps story arcs later.

So, the RFE is this:
It appears facilities have a "local" modifier for their particular facility if a battle takes place at the facility- e.g. Mech Garrison for a Mech Base.  And a "shared" modifier that they apply to battles fought on the track -- e.g. Mech Reinforcements.

So the request is a similar thing for units.  

1.  Have an "individual" modifier and/or stat section that you can apply to just that individual/force and a "shared" modifier that you can attach to a force that applies to all other forces on its team.  I did try to add a shared modifier to a unit but got a NPE.  
2.  Have the option to manually/randomly assign edge and enable triggers and manually/randomly assign SPAs in the same way.
3. Have a flag to "generate portrait" and "generate individual camo" for this specific force. (with perhaps folder pointers).
4. Allow pre and  post force modifiers to be linked.  I tried to create one that had both +skill and +quality but it did not seem to work.

I think this will fill in the last gap in what MHQ can generate.  This is more random than generating from a fixed mul, but less random than the current method, and would allow for creation of random-but-still-somewhat-unique commanders, aces, villains, vip assassination targets, bodyguards, modifiers, etc.

Thank you.

Longer Context.: So, Ive just done some simple .xml edits to add some new battle modifiers to stratcon.  Basically just existing ones upgraded.  Zero code changes.  Im still testing them, but they work, with some exceptions:

I can get new units to appear when and where I want, and I can get +equipment quality globally, and I can get +skill level globally.  But I dont seem to be able to get +quality and skill at the same time (I think because one is generated pre-force and one is generated post-force), nor can I get +quality and/or skill to just a specific unit (as opposed to all of them).  Ive read all the documentation I can find on it, and looked at the code, and it appears neither of those is possible right now.  If I misunderstand, or am missing it, please let me know.

So I will explain by way of example.  I created an "Enemy Mech Ace" modifier.  I can get him to show up when I want.  But I tried to give him skill +1 and quality +1 to just his individual unit (basically copying "Veterans" and "Good Equipment" onto him).  I also tried the same thing with an Enemy Mech Officer but added the BV increase modifier.  Neither seemed to work.  (I was able to create Elites and Superior Equipment modifiers and they do work when done by themselves).

1. This did not apply the bonus to him, whether I generated him pre or post generation, or whether I changed his generation order to 1 or 6.  There does not seem to be a way to unhitch his individual force skill or equipment level from the the rest of the entire bot team.  Either all of them get a bonus or none of them do. 
2. This did not (appear) to apply the bonus to his team units, no matter the time or location of his generation, unlke if I just added Veterans or Good Equipment manually.

Ive looked at the code in the Dynamic Scenario Factory and found where it generally happens, I think, but Its not something Im qualified to edit or expand on.

Thank you.
